### DISCUSSION TITLE
validating config presets

### DISCUSSION THREAD

**Discussion by amimas** [renovatebot/renovate#13649] [Request Help]
I'm creating a separate repository for hosting my config presets. I wanted some way of validating the presets to make sure contributing changes into the preset do not lead to any issues due to incorrect config.

I'm trying to use the `renovate-config-validator` cli but regardless of what I pass to the file, it always returns a message saying:

```
 INFO: Config validated successfully
✨  Done in 3.26s.
```

Here are couple of preset files I tried:

`my-preset.json`:

```json
{
  "blah": "blah",
  "automerge": "never",
}
```

`my-preset.js`:

```js
module.exports = {
  foo: "bar",
}
```

It seems like the config validator does not handle presets config. Most importantly, it incorrectly returns message about the config being valid. Am I doing something wrong here?

**Comment by amimas** [comment]
I just managed to get the validator working successfully. Originally I was running the validator like this: `renovate-config-validator path/to/preset-file.json`. This approach didn't seem to work as explained above.

I tried setting the environment variable instead:

```
$ export RENOVATE_CONFIG_FILE=presets/my-preset.json
$ renovate-config-validator
 INFO: Validating presets/my-preset.json
ERROR: presets/my-preset.json contains errors
       "errors": [
         {
           "topic": "Configuration Error",
           "message": "Configuration option `automerge` should be boolean. Found: \"never\" (string)"
         },
         {
           "topic": "Configuration Error",
           "message": "Invalid configuration option: blah"
         }
       ]
error Command failed with exit code 1.
```

Not sure why passing the file directly as a parameter to the CLI didn't work. I'm using renovate v31.19.0

**Comment by viceice** [comment]
Checkout our preset validation at https://github.com/renovatebot/.github/blob/main/.github/workflows/test.yml

**Reply by amimas** [reply]
Thanks. Looks like you're validating presets the same way; by setting `RENOVATE_CONFIG_FILE` environment variable. So, I guess the config validator never takes a file as an input, right?

I'm using the validator in my self-hosted bot where I use it like this: `renovate-config-validator config.js`. This works, but I think it only works because the CLI looks for `config.js` by default. So, to me it seems like the CLI doesn't take any positional parameter as input.

It'd be nice if the CLI also has some help. For example, running `renovate-config-validator --help` doesn't do anything. It just returns a message saying config validated successfully. This is probably another bug/missing-feature.

**Reply by rarkins** [reply]
PRs welcome
